src/Microsoft.AspNetCore.Mvc.Analyzers/TopLevelParameterAttributeAnalyzer.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.AspNetCore.Mvc.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class TopLevelParameterAttributeAnalyzer : DiagnosticAnalyzer
```

Comment:
Are you planning a separate analyzer or analyzer pair for `[Required]` on complex types? #7415 covers that case too.

Reply:
Yes I'm planning seperate analyzer for that.

Reply:
We'll writer analyzer as well  in this issue. If it require please update issue.

Reply:
Sorry, i mean codefixer.

Reply:
Not doing fixers is fine for this PR. As I said earlier,
> Initial priority is covering #7415 in full, especially covering all top-level models and providing correct advice. Well, the "might be worthwhile" point is optional.

Reply:
OK, ☺️

Reply:
Could we call this `RequiredAttributeAnalyzer`? There's already an analyzer by the same name in the Api.Analyzers package